node-graph/gcore/src/vector/vector_nodes.rs

```diff
@@ -1715,6 +1715,29 @@ fn bevel(_: impl Ctx, source: VectorDataTable, #[default(10.)] distance: Length)
 	result
 }
 
+#[node_macro::node(name("Close Path"), category("Vector"), path(graphene_core::vector))]
+fn close_path(_: impl Ctx, source: VectorDataTable) -> VectorDataTable {
+	let mut new_table = VectorDataTable::empty();
+	for mut source_instance in source.instance_iter() {
+		source_instance.instance.close_subpaths();
+		new_table.push(Instance {
+			instance: source_instance.instance,
+			transform: source_instance.transform,
+			alpha_blending: source_instance.alpha_blending,
+			source_node_id: source_instance.source_node_id,
+		})
```

Comment:
You don't need to construct a new instance composed of all the same fields as the current instance. Just push the current instance (`source_instance`).